File: Data Structures & Algorithms/longest-common-prefix/submission-3.go
func longestCommonPrefix(strs []string) string {
    for i := 0; i < len(strs[0]); i++ {
        c := strs[0][i]
        for _, word := range strs {
            if i >= len(word) || c != word[i] {
                return strs[0][:i]
            }
        }
    }
    return strs[0]
}

File: Data Structures & Algorithms/string-encode-and-decode/submission-3.py
class Solution:

    def encode(self, strs: List[str]) -> str:
        res = []
        for word in strs:
            res.append(f"{len(word)}#{word}")
        return "".join(res)

    def decode(self, s: str) -> List[str]:
        res = []
        i = 0
        while i < len(s):
            # get num
            j = i
            while s[j] != "#":
                j += 1
            num = int(s[i:j])
            # get word
            start = j + 1
            end = start + num
            res.append(s[start:end])
            i = end
        return res
